### PR TITLE
JENKINS-47490 Make the step indicator animate

### PR DESCRIPTION
**Pull request by NicuPascu**
Make the step indicator animate again (used to rely on css animation that didn't work on IE and was removed)

# Description

See [JENKINS-47490](https://issues.jenkins-ci.org/browse/JENKINS-47490).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [ ] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [x] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

### CODE REVIEW

**Comment by sophistifunk** [comment]
Why not change it to use `<StatusIndicator/>` instead of `getGlyphFor()`? Duplicating the animation logic (and detection of when it should spin) in here doesn't seem great.

**Comment by NicuPascu** [comment]
@sophistifunk I just wanted to fix the missing animation, I didn't want to refactor the component because of this. TBH I don't know why the component doesn't use <StatusIndicator /> as you suggested, I'll try to do that and see if anything breaks.

**Comment by michaelneale** [comment]
@NicuPascu any thoughts on how to test this (automated)? probably an ATH thing?

**Comment by sophistifunk** [comment]
@michaelneale you kiss you mother with that mouth?

**Comment by i386** [comment]
@NicuPascu I think an enzyme test to ensure the right component (indeterminate progress) gets put in the ResultItem when it is "running" would be important to the fix here. As a general rule, we should be adding tests like these to ensure we don't get a future regression.

If we were writing a new component, I'd expect these sorts of tests would be written before the component would be considered done.

**Comment by michaelneale** [comment]
yeh probably not an ATH thing - but this has regressed, so where there is a regression,there is probably a missing test.

**Comment by sophistifunk** [comment]
That should definitely be a unit test, not an ATH test

**Comment by michaelneale** [comment]
@sophistifunk agree

**Comment by sophistifunk** [comment]
Looks good, given the curry @NicuPascu got trying to re-use the existing component I suggested. Next time I need to make changes to `ResultItem` I might have a timeboxed crack at it, but it's not a huge (or super complex) duplication so it's no biggie.

bb-tf

File: jenkins-design-language/src/js/components/ResultItem.jsx [hunk 1]
```diff
@@ -109,12 +134,21 @@ export class ResultItem extends Component {
             },
         };
 
+        if (resultClean === "running" && !this.infiniteRotationRunning) {
+            requestAnimationFrame(this.infiniteLoadingTimer);
```
Comment (scherler) [inline jenkins-design-language/src/js/components/ResultItem.jsx]: Should that not better go into `componentDidMount` and validate it in `componentWillReceiveProps`?

**Comment by NicuPascu** [comment]
@scherler after a discussion with @sophistifunk we decided that the code is more readable in this form, is there a reason I might be missing of why it would be worth it to do it like that?

**Comment by scherler** [comment]
http://busypeoples.github.io/post/react-component-lifecycle/ the render method comes before didMount and generates the DOM that is then fully rendered in didMount https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame says:

> You should call this method whenever you're ready to update your animation onscreen.

That is only th case when your DOM has finished rendering hence adding it to render is wrong.

**Comment by NicuPascu** [comment]
Right, but that statement assumes we want to manipulate the DOM before the next browser re-paint, in our case we do not change the DOM in any way, we just increment the rotation degrees variable that is being used in the render function. Even if the first render would show the icon rotated by a few degrees (which is not the case), it would not be noticeable to the human eye, we rotate the icon 60 times per second.

**Comment by sophistifunk** [comment]
I vote for leaving it with the simplified version for now, and if it becomes a performance issue we deal with it more cleanly at that point. React should deal with the single attr change without any problem, and since we're using `requestAnimationFrame` the browser will self-throttle the animation if we're causing any sort of a ruckus.

**Comment by michaelneale** [comment]
Ruckus is a word not used enough: https://www.youtube.com/watch?v=hR9F8t39Az4

**Pull request merged**